src/proto/connection.rs

```diff
@@ -282,7 +282,13 @@ where
                 State::Closing(reason, initiator) => {
                     tracing::trace!("connection closing after flush");
                     // Flush/shutdown the codec
-                    ready!(self.codec.shutdown(cx))?;
+                    if let Err(e) = ready!(self.codec.shutdown(cx)) {
+                        // If the error kind is NotConnected then ignore that
+                        // since it means the connection is already shutdown.
```

Comment:
nit: maybe:
```suggestion
                        // If the error kind is NotConnected, ignore it, since
                        // it just means the connection is already shutdown.
```